vnet/net.go

```diff
@@ -353,24 +374,21 @@ func (v *vNet) hasIPAddr(ip net.IP) bool {
 	return false
 }
 
+// caller must hold the mutex
 func (v *vNet) allocateLocalAddr(ip net.IP, port int) error {
+	// gather local IP addresses to bind
 	var ips []net.IP
-
-	switch ip.String() {
-	case "0.0.0.0":
+	if ip.IsUnspecified() {
```

Comment:
before in the case of "::" getAllIPAddrs will get ipv6, now that has changed, is that a problem?